### PR TITLE
yder: 1.4.6 -> 1.4.7

### PR DESCRIPTION
**Pull request by JohnAZoidberg**
###### Motivation for this change
- https://github.com/babelouest/yder/releases/tag/v1.4.7
- https://github.com/babelouest/orcania/releases/tag/v2.0.1

Now we don't need the custom patch anymore to make it build on Darwin. Please someone test or have @GrahamcOfBorg test it :)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by srhb** [comment]
@GrahamcOfBorg build orcania yder

**Comment by srhb** [comment]
Ruh-oh, looks like it failed on Darwin.

File: pkgs/development/libraries/yder/fix-darwin.patch [hunk 1]
```diff
@@ -1,159 +0,0 @@
...
--            set(LIBS ${LIBS} ${SUBUNIT_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} m rt)
-+            set(LIBS yder ${LIBS} ${SUBUNIT_LIBRARIES} ${ORCANIA_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${CHECK_LIBRARIES} m)
-         endif ()
- 
-         set(TESTS yder_test)
-@@ -186,7 +185,6 @@ if (BUILD_YDER_TESTING)
-             target_include_directories(${t} PUBLIC ${TST_DIR})
-             target_link_libraries(${t} PUBLIC ${LIBS})
-             add_test(NAME ${t}
--                    WORKING_DIRECTORY ${TST_DIR}
```
Comment (JohnAZoidberg) [inline pkgs/development/libraries/yder/fix-darwin.patch]: Looks like this line of my patch didn't make it upstream

**Comment by JohnAZoidberg** [comment]
Oh no :(
Can anyone with MacOS please help me debug this?
I think the error means that the test execution cannot find the library at runtime - so I think patching that `WORKING_DIRECTORY ${TST_DIR}` line away should probably fix it.

**Comment by mmahut** [comment]
@GrahamcOfBorg build orcania yder

**Comment by JohnAZoidberg** [comment]
Nice, it works!
It's not the prettiest solution but I think fixing it would need a rewrite of upstream's integration of their tests with Cmake. Not sure yet how.
`DYLD_FALLBACK_LIBRARY_PATH` might not work with MacOS' [System Integrity Protection](https://developer.apple.com/library/archive/documentation/Security/Conceptual/System_Integrity_Protection_Guide/RuntimeProtections/RuntimeProtections.html)

**Pull request merged**